File: Motocitizen/src/motocitizen/user/Role.java
package motocitizen.user;

public enum Role {
    RO("readonly", "только чтение"),
    BANNED("banned", "забанен"),
    STANDARD("standart", "пользователь"),
    MODERATOR("moderator", "модератор"),
    ADMINISTRATOR("admin", "администратор"),
    DEVELOPER("developer", "разработчик");

    private final String code;
    private final String text;

    Role(String code, String text) {
        this.code = code;
        this.text = text;
    }

    public static Role parse(String role) {
        for (Role a : Role.values()) {
            if (a.code.equals(role)) return a;
        }
        return RO;
    }

    public boolean isStandard() {
        return this.compareTo(STANDARD) >= 0;
    }

    public boolean isModerator() {
        return this.compareTo(MODERATOR) >= 0;
    }

    public String getName() {
        return this.text;
    }

    public String getCode() {
        return code;
    }

}

File: Motocitizen/src/motocitizen/user/Role.kt
package motocitizen.user

import motocitizen.dictionary.Dictionary

enum class Role constructor(override val code: Int, override val text: String) : Dictionary<Int> {
    RO(0, "только чтение"),
    STANDARD(1, "пользователь"),
    MODERATOR(2, "модератор"),
    DEVELOPER(3, "разработчик");
}
